experimental/dds/tree/src/test/HistoryEditFactory.tests.ts

```diff
@@ -120,5 +120,28 @@ describe('revert', () => {
 				).to.be.undefined;
 			});
 		});
+		describe('when reverting the insert/detach of an empty trait', () => {
+			it('when reverting a insert of an empty trait', () => {
+				const insertedNodeId = 0 as DetachedSequenceId;
+				const insertedBuild = ChangeInternal.build([], insertedNodeId);
+				const insertChange = ChangeInternal.insert(insertedNodeId, {
+					referenceTrait: testTree.left.traitLocation,
+					side: Side.After,
+				});
+				const result = revert([insertedBuild, insertChange], testTree.view);
+				expect(result).to.be.undefined;
+			});
+			it('when reverting a detach of an empty trait', () => {
+				const insertedNodeId = 0 as DetachedSequenceId;
```

Comment:
why are the build/insert changes needed here? I would have expected this to simply be a call to revert with a ChangeInternal.detach(StableRangeInternal.all({label: testTreesomeTraitIdThatDoesntExist, parent: anyIdInTheTree}), someDetachedId).

I might be misunderstanding the bug, though :)

Reply:
Thank you for your quick feedback :)

I have made the changes to the test case using some test label that doesn't exist on the tree 'testTraitLabel', and is getting the correct result `undefined`, but due to an invalid `detach.source` (and not because it is an empty trait).

Would this be the expected behavior, or should the test be updated to test a trait that exists in the tree (but is empty)?